{"kind":"symbol","identifier":{"interfaceLanguage":"swift","url":"doc:\/\/com.apptentive.Apptentive\/documentation\/ApptentiveKit\/apptentiveAssertionHandler"},"abstract":[{"text":"The method to call when a critical error occurs.","type":"text"}],"schemaVersion":{"patch":0,"major":0,"minor":3},"sections":[],"metadata":{"externalID":"s:13ApptentiveKit26apptentiveAssertionHandleryySSyXK_s12StaticStringVSutcvp","symbolKind":"var","roleHeading":"Global Variable","fragments":[{"text":"var","kind":"keyword"},{"text":" ","kind":"text"},{"kind":"identifier","text":"apptentiveAssertionHandler"},{"text":": (() -> ","kind":"text"},{"text":"String","kind":"typeIdentifier","preciseIdentifier":"s:SS"},{"text":", ","kind":"text"},{"kind":"typeIdentifier","text":"StaticString","preciseIdentifier":"s:s12StaticStringV"},{"kind":"text","text":", "},{"kind":"typeIdentifier","preciseIdentifier":"s:Su","text":"UInt"},{"kind":"text","text":") -> ()"}],"title":"apptentiveAssertionHandler","modules":[{"name":"ApptentiveKit"}],"role":"symbol"},"hierarchy":{"paths":[["doc:\/\/com.apptentive.Apptentive\/documentation\/ApptentiveKit"]]},"primaryContentSections":[{"declarations":[{"tokens":[{"text":"var","kind":"keyword"},{"text":" ","kind":"text"},{"kind":"identifier","text":"apptentiveAssertionHandler"},{"kind":"text","text":": ("},{"kind":"attribute","text":"@autoclosure "},{"kind":"text","text":"() -> "},{"preciseIdentifier":"s:SS","text":"String","kind":"typeIdentifier"},{"text":", ","kind":"text"},{"preciseIdentifier":"s:s12StaticStringV","text":"StaticString","kind":"typeIdentifier"},{"text":", ","kind":"text"},{"text":"UInt","preciseIdentifier":"s:Su","kind":"typeIdentifier"},{"kind":"text","text":") -> ()"}],"languages":["swift"],"platforms":["iOS"]}],"kind":"declarations"},{"content":[{"type":"heading","text":"Discussion","level":2,"anchor":"discussion"},{"type":"paragraph","inlineContent":[{"text":"This can be overriden, for example:","type":"text"}]},{"code":["apptentiveAssertionHandler = { message, file, line in","    print(\"\\(file):\\(line): Apptentive critical error: \\(message())\")","}"],"type":"codeListing","syntax":null}],"kind":"content"}],"variants":[{"traits":[{"interfaceLanguage":"swift"}],"paths":["\/documentation\/apptentivekit\/apptentiveassertionhandler"]}],"references":{"doc://com.apptentive.Apptentive/documentation/ApptentiveKit/apptentiveAssertionHandler":{"role":"symbol","url":"\/documentation\/apptentivekit\/apptentiveassertionhandler","identifier":"doc:\/\/com.apptentive.Apptentive\/documentation\/ApptentiveKit\/apptentiveAssertionHandler","fragments":[{"kind":"keyword","text":"var"},{"text":" ","kind":"text"},{"kind":"identifier","text":"apptentiveAssertionHandler"},{"text":": (() -> ","kind":"text"},{"preciseIdentifier":"s:SS","text":"String","kind":"typeIdentifier"},{"kind":"text","text":", "},{"text":"StaticString","kind":"typeIdentifier","preciseIdentifier":"s:s12StaticStringV"},{"text":", ","kind":"text"},{"text":"UInt","kind":"typeIdentifier","preciseIdentifier":"s:Su"},{"text":") -> ()","kind":"text"}],"abstract":[{"type":"text","text":"The method to call when a critical error occurs."}],"title":"apptentiveAssertionHandler","kind":"symbol","type":"topic"},"doc://com.apptentive.Apptentive/documentation/ApptentiveKit":{"identifier":"doc:\/\/com.apptentive.Apptentive\/documentation\/ApptentiveKit","type":"topic","url":"\/documentation\/apptentivekit","abstract":[],"kind":"symbol","role":"collection","title":"ApptentiveKit"}}}